src/package-linker.js

```diff
@@ -446,6 +446,21 @@ export default class PackageLinker {
       }
       const ref = pkg._reference;
       invariant(ref, 'Package reference is missing');
+      // TODO: We are taking the "shortest" ref tree but there may be multiple ref trees with the same length
+      const refTree = ref.requests.map(req => req.parentNames).sort((arr1, arr2) => arr1.length - arr2.length)[0];
+
+      const getLevelDistance = pkgRef => {
+        let minDistance = Infinity;
+        for (const req of pkgRef.requests) {
+          const distance = refTree.length - req.parentNames.length;
+
+          if (distance >= 0 && distance < minDistance && req.parentNames.every((name, idx) => name === refTree[idx])) {
```

Comment:
It might be slightly less readable but more efficient to use `!some` to exit early if they don't match

Reply:
I think both `every` and `some` do early exits so don't think it would change perf.

Reply:
ugh, I keep forgetting that sorry!